### PR TITLE
Add delete attribute features

### PR DESCRIPTION
**Pull request by eknuviad**
Includes complete implementation of add and delete attributes which can be reviewed and merged.
line implementation will follow soon.

### CODE REVIEW

File: domain-model-assistant/Assets/Components/Scripts/Position.cs [hunk 1]
```diff
@@ -27,3 +27,22 @@ public class CreateJson
 
     public float y; 
 }
+
+
+public class AddAttributeInfo
```
Comment (YounesB-McGill) [inline domain-model-assistant/Assets/Components/Scripts/Position.cs:32]: `...Info` is a software engineering term

File: domain-model-assistant/Assets/Components/Scripts/Section.cs [hunk 2]
```diff
@@ -98,4 +99,4 @@ public void AddAttribute(string _id, string name, string type)
         this.AddTextBox(TB);
     }
 
-}
+}
```
Comment (YounesB-McGill) [inline domain-model-assistant/Assets/Components/Scripts/Section.cs:102]: Missing newline at end of file

File: domain-model-assistant/Assets/Components/Scripts/TextBox.cs [hunk 3]
```diff
@@ -13,16 +15,38 @@ public string ID
...
     {
 
     }
 
     void Update()
     {
+        if(Input.GetKeyDown(KeyCode.Return)){
+            Debug.Log("Enter");
+            string _id = this.ID;
+             if(isValid() && _id.Equals("-1")){
```
Comment (YounesB-McGill) [inline domain-model-assistant/Assets/Components/Scripts/TextBox.cs:42]: Formatting here does not match the default C# Style used in the rest of the codebase

File: domain-model-assistant/Assets/Components/Scripts/TextBox.cs [hunk 4]
```diff
@@ -34,19 +58,72 @@ void Update()
...
-    public bool SetSection(GameObject section)
+    public bool isValid(){
+        bool res = false;
+        string text = this.GetComponent<InputField>().text;
+        //check that inputfield is of a particular format (Eg. int year)
+        string[] values = text.Split(' ');
+        Debug.Log(values.Length);
+        if (values.Length == 2 && !String.IsNullOrWhiteSpace(values[1]))
+        {
+            // if(!values[1].Equals(' ')
```
Comment (YounesB-McGill) [inline domain-model-assistant/Assets/Components/Scripts/TextBox.cs:69]: Dead code :skull:

File: domain-model-assistant/Assets/Components/Scripts/TextBox.cs [hunk 5]
```diff
@@ -34,19 +58,72 @@ void Update()
...
+            this.typeId = Int16.Parse(tmpTypeId);
+        } 
+    }
+
+    public int GetTypeId()
+    {
+        return this.typeId;
+    }
+        
+    public bool SetSection(GameObject sSection)
```
Comment (YounesB-McGill) [inline domain-model-assistant/Assets/Components/Scripts/TextBox.cs:114]: Why is the param name `sSection`? What does the first "s" stand for?

File: domain-model-assistant/Assets/Components/Scripts/TextBox.cs [hunk 6]
```diff
@@ -34,19 +58,72 @@ void Update()
...
         return true;
     }
 
     public GameObject GetSection()
     {
-        return section;
+        return this.section;
     }
 
     public void OnBeginHold2()
```
Comment (YounesB-McGill) [inline domain-model-assistant/Assets/Components/Scripts/TextBox.cs:129]: Why is there a "2" at the end of this method name? Rename this or at least add a comment to make this clear

File: domain-model-assistant/Assets/Components/Scripts/TextBox.cs [hunk 7]
```diff
@@ -69,6 +146,7 @@ public void OnEndHold2()
 
     void SpawnAttributeCross()
     {
+        Debug.Log("here");
```
Comment (YounesB-McGill) [inline domain-model-assistant/Assets/Components/Scripts/TextBox.cs:149]: Useless print statement. Make this more specific, eg, "In Textbox.SpawnAttributeCross()" or remove it

**Review by YounesB-McGill — COMMENTED** [review]
Please see my comments on this feature. Addressing these comments will make PR #80 easier to review, since its first commits are also here

File: domain-model-assistant/Assets/Components/Scripts/Diagram.cs [hunk 8]
```diff
@@ -310,16 +318,71 @@ public void UpdateNames()
...
+            string jsonData = JsonUtility.ToJson(info);
+            Debug.Log(jsonData);
+            // @param body {"rankIndex": Integer, "typeId": Integer, "attributeName": String}
+            PostRequest(AddAttributeEndpoint + "/" + "class" + "/" + _id + "/" + "attribute", jsonData);
+            Debug.Log(AddAttributeEndpoint + "/" + "class" + "/" + _id + "/" + "attribute");
+            reGetRequest = true;
+            GetRequest(GetCdmEndpoint);
+        }
+    }
+    /// Deletes Attribute
```
[the remark this answers is not in the document]
Comment (violetwei) [reply domain-model-assistant/Assets/Components/Scripts/Diagram.cs:356]: Fixed

File: domain-model-assistant/Assets/Components/Scripts/Diagram.cs [hunk 9]
```diff
@@ -310,16 +318,71 @@ public void UpdateNames()
...
+    /// </summary> 
+    public void DeleteAttribute(GameObject textBox)
+    {
+        // Debug.Log("We entered into delete attr:" + textBox.GetComponent<TextBox>().text);
```
[the remark this answers is not in the document]
Comment (violetwei) [reply domain-model-assistant/Assets/Components/Scripts/Diagram.cs:360]: Fixed

File: domain-model-assistant/Assets/Components/Scripts/Diagram.cs [hunk 10]
```diff
@@ -310,16 +318,71 @@ public void UpdateNames()
...
+            string _id = textBox.GetComponent<TextBox>().ID;
+            DeleteRequest(DeleteAttributeEndpoint, _id);
+            reGetRequest = true;
+            GetRequest(GetCdmEndpoint);
+            // No need to remove or destroy the node here since entire class diagram is recreated
+        }
+        else
+        {
+            //RemoveNode(section);
+            //Destroy(node);
```
[the remark this answers is not in the document]
Comment (violetwei) [reply domain-model-assistant/Assets/Components/Scripts/Diagram.cs:372]: Resolved

File: domain-model-assistant/Assets/Components/Scripts/Position.cs (hunk 1 above)
Comment (violetwei) [reply domain-model-assistant/Assets/Components/Scripts/Position.cs:32]: Renamed the class

File: domain-model-assistant/Assets/Components/Scripts/Section.cs (hunk 2 above)
Comment (violetwei) [reply domain-model-assistant/Assets/Components/Scripts/Section.cs:102]: Done

File: domain-model-assistant/Assets/Components/Scripts/TextBox.cs (hunk 3 above)
Comment (violetwei) [reply domain-model-assistant/Assets/Components/Scripts/TextBox.cs:42]: Fixed

File: domain-model-assistant/Assets/Components/Scripts/TextBox.cs (hunk 4 above)
Comment (violetwei) [reply domain-model-assistant/Assets/Components/Scripts/TextBox.cs:69]: Removed

File: domain-model-assistant/Assets/Components/Scripts/TextBox.cs (hunk 5 above)
Comment (violetwei) [reply domain-model-assistant/Assets/Components/Scripts/TextBox.cs:114]: Renamed the param here

File: domain-model-assistant/Assets/Components/Scripts/TextBox.cs (hunk 6 above)
Comment (violetwei) [reply domain-model-assistant/Assets/Components/Scripts/TextBox.cs:129]: Fixed

File: domain-model-assistant/Assets/Components/Scripts/TextBox.cs (hunk 7 above)
Comment (violetwei) [reply domain-model-assistant/Assets/Components/Scripts/TextBox.cs:149]: Done

File: domain-model-assistant/Assets/Components/Scripts/Diagram.cs [hunk 11]
```diff
@@ -310,16 +318,71 @@ public void UpdateNames()
...
     public void ResetDiagram()
     {
         foreach (var comp in compartmentedRectangles)
         {
-            // popup menu is destroyed in comp rect class when delete is called
-            // we only need to destroy the attributes.
-            // get first section, loop through all attributes, destroy any attribute cross objects
+            //popuup menu is destroyed in comp rect class whenn delete is called
+            //we only need to destroy the  atriibutes.
+            //get first section, loop through all attributes, destroy any attribute cross objects
```
[the remark this answers is not in the document]
Comment (YounesB-McGill) [reply domain-model-assistant/Assets/Components/Scripts/Diagram.cs]: Reapplied

**Review by YounesB-McGill — APPROVED** [review]
Approving and merging this now :boat:  Please merge the updated main branch into the CapstoneDemo branch, which currently under review in PR #80.

**Pull request merged**